frontend/src/components/dashboard/settings/api-key/ApiKey.tsx

```diff
@@ -1,155 +1,127 @@
-      break
-    case ApiKeyState.COPIED:
-      buttonLabel = 'Copied'
-      buttonIcon = 'bx-check'
-      buttonClass = styles.blueButton
-      break
-    case ApiKeyState.REGENERATE:
-      buttonLabel = 'Regenerate'
-      buttonIcon = 'bx-refresh'
-      buttonClass = styles.blueButton
-      break
-    default:
-      break
+  function openDeleteModal(keyId: string) {
+    const apiKey = apiKeys.find((k) => k.id === keyId)
+    if (!apiKey) return
+    modalContext.setModalContent(
+      <ConfirmModal
+        title={`Are you sure you want to delete API key "${apiKey.label}"?`}
+        subtitle="Deleting your API key is irreversible, and any integrations using this key will need to be updated with a new API key to continue working."
+        onConfirm={() => {
+          removeApiKey(keyId)
+          void deleteApiKey(keyId)
+        }}
+        onCancel={() => modalContext.close()}
+        buttonText="Confirm delete"
+        buttonIcon="bx-trash"
+        cancelText="Cancel"
+        destructive
+      />
+    )
   }
 
-  return (
+  return isPageLoading ? (
+    <div className={styles.loader}>
+      <i className="bx bx-loader-alt bx-spin" />
+    </div>
+  ) : (
     <>
-      <StepHeader title="API Key">
-        <p className={styles.helpText}>
-          After generating your API key, please make a copy of it immediately as
-          it will only be shown once. Upon leaving or refreshing this page, the
-          key will be hidden.
-        </p>
-      </StepHeader>
-      <TextInputWithButton
-        value={apiKey}
-        onChange={() => {
-          return
-        }}
-        onClick={onButtonClick}
-        className={buttonClass}
-        textRef={apiKeyRef}
-        errorMessage={errorMsg}
-        buttonLabel={
-          <>
-            {buttonLabel} API key
-            <i className={cx('bx', buttonIcon)} />
-          </>
-        }
-      />
+      <div className={styles.header}>
+        <h3>API Key</h3>
+        <PrimaryButton
+          className={styles.blueButton}
+          onClick={openGenerateModal}
+        >
+          <i className="bx bx-plus"></i>
+          Generate API Key
+        </PrimaryButton>
+      </div>
+      <p className={styles.helpText}>
+        Creating a new API key gives you access to our platform's data and
+        features, and allows you to build custom integrations, automate tasks,
+        and streamline workflows. It also helps keep your data more secure.
+      </p>
+      {apiKeys.length > 0 && (
+        <table className={styles.apiKeyTable}>
+          <thead>
+            <tr>
+              <th className="lg">Key name</th>
+              <th className="sm">Last 5 Digits</th>
+              <th className="sm"></th>
+            </tr>
+          </thead>
+          <tbody>
+            {apiKeys.map((k) => (
+              <tr key={k.id}>
+                <td className="lg">{k.label}</td>
+                <td className="sm">••••• {k.lastFive}</td>
```

Comment:
**[question]** just curious, the space between the `•` and the `{k.lastFive}` is an aesthetic decision?

this is how it currently looks:

<img width="760" alt="image" src="https://user-images.githubusercontent.com/67887489/229529747-bd5e208d-32b2-4a3f-8a8a-7c9707a61e33.png">


this is how it looks without the space:

<img width="762" alt="image" src="https://user-images.githubusercontent.com/67887489/229529519-2c9c33b9-dfd0-4691-a36a-5fe0493cc0e5.png">